### PR TITLE
[feature](jni) support complex types in jni framework

### PR DESCRIPTION
**Pull request by AshinGau**
## Proposed changes

Support complex types in jni framework, and successfully run end-to-end on hudi.
### How to Use
Other scanners only need to implement three interfaces in `ColumnValue`:
```
// Get array elements and append into values
void unpackArray(List<ColumnValue> values);

// Get map key array&value array, and append into keys&values
void unpackMap(List<ColumnValue> keys, List<ColumnValue> values);

// Get the struct fields specified by `structFieldIndex`, and append into values
void unpackStruct(List<Integer> structFieldIndex, List<ColumnValue> values);
```
Developers can take `HudiColumnValue` as an example.

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by AshinGau** [comment]
run buildall

**Comment by AshinGau** [comment]
run buildall

**Review by wuwenchi — APPROVED** [review]

File: be/src/vec/exec/jni_connector.cpp [hunk 1]
```diff
@@ -301,13 +305,82 @@ Status JniConnector::_fill_column(ColumnPtr& doris_column, DataTypePtr& data_typ
...
+    // offsets are stored as int type in java side
+    int* offsets = reinterpret_cast<int*>(_next_meta_as_ptr());
+    size_t origin_size = offsets_data.size();
+    offsets_data.resize(origin_size + num_rows);
+    size_t start_offset = offsets_data[origin_size - 1];
+    for (size_t i = 0; i < num_rows; ++i) {
+        offsets_data[origin_size + i] = offsets[i] + start_offset;
+    }
+
+    // offsets[num_rows - 1] == offsets_data[origin_size + num_rows - 1] - start_offset
```
Comment (DongLiang-0) [inline be/src/vec/exec/jni_connector.cpp:338]: Can this line of code be deleted?

**Comment by AshinGau** [comment]
run buildall

**Comment by AshinGau** [comment]
run buildall

**Review by Mryange — APPROVED** [review]

**Review by zy-kkk — APPROVED** [review]

**Pull request merged**